### PR TITLE
Fix importing multiple accounts from ledger

### PR DESCRIPTION
**Pull request by buberdds**
Select the first wallet from a selection after all wallets are loaded

Closes https://github.com/oasisprotocol/oasis-wallet-web/issues/934

### CODE REVIEW

File: src/app/state/wallet/saga.test.ts [hunk 1]
```diff
@@ -70,6 +71,40 @@ describe('Wallet Sagas', () => {
...
+            isOpen: true,
+            selectedWallet: 0,
+            wallets: [
+              {
+                address: 'oasis1qz0k5q8vjqvu4s4nwxyj406ylnflkc4vrcjghuwk',
+                id: 0,
+              },
+              {
+                address: 'oasis1qq2vzcvxn0js5unsch5me2xz4kr43vcasv0d5eq4',
+                id: 1,
```
Comment (lukaw3d) [inline src/app/state/wallet/saga.test.ts:88]: Does this need to be here to work around `walletId` global variable messing everything up?

**Comment by lukaw3d** [comment]
How come this fixed the bug?

File: src/app/state/wallet/saga.test.ts [hunk 2]
```diff
@@ -70,6 +71,40 @@ describe('Wallet Sagas', () => {
         .silentRun(50)
     })
 
+    it('Should open from ledger and select the first imported wallet as active', () => {
```
Comment (lukaw3d) [inline src/app/state/wallet/saga.test.ts:74]: Can this test be extended to validate the bug was fixed?

**Comment by lukaw3d** [comment]
Update: I can't reproduce the bug with expectSaga on master. Looks like reducer + saga scheduling in app is different than in expectSaga.

```ts
const result = await expectSaga(rootWalletSaga)
.provide(providers)
.withReducer(walletReducer)
.dispatch(
  walletActions.openWalletsFromLedger([
    {
      address: 'oasis1qq2vzcvxn0js5unsch5me2xz4kr43vcasv0d5eq4',
    } as LedgerAccount,
    {
      address: 'oasis1qq5t7f2gecsjsdxmp5zxtwgck6pzpjmkvc657z6l',
    } as LedgerAccount,
    {
      address: 'oasis1qrusst9f8ey6kphgp5d02nuf4kx5qsa80vamtxj8',
    } as LedgerAccount,
    {
      address: 'oasis1qpqs54pdwxstpjefw5t6ucy425nt4yud3c4fgr8p',
    } as LedgerAccount,
    {
      address: 'oasis1qrf54xw003wkznm90w0q05ns8nryla0455tlex38',
    } as LedgerAccount,
  ]),
)
.fork(walletSaga)
.put({ type: walletActions.walletSelected.type, payload: 4 })
.silentRun(50)

expect(result.storeState).toMatchObject({
  selectedWallet: 4,
} as WalletState)
```

put(selectWallet) in step 8 schedules walletSelected at the end:
```md
1. 'wallet/addWallet', payload: { id: 4, address: 'oasis1qq2vzcvxn0js5unsch5me2xz4kr43vcasv0d5eq4' }
2. 'wallet/addWallet', payload: { id: 5, address: 'oasis1qq5t7f2gecsjsdxmp5zxtwgck6pzpjmkvc657z6l' }
3. 'wallet/addWallet', payload: { id: 6, address: 'oasis1qrusst9f8ey6kphgp5d02nuf4kx5qsa80vamtxj8' }
4. 'wallet/addWallet', payload: { id: 7, address: 'oasis1qpqs54pdwxstpjefw5t6ucy425nt4yud3c4fgr8p' }
5. 'wallet/addWallet', payload: { id: 8, address: 'oasis1qrf54xw003wkznm90w0q05ns8nryla0455tlex38' }

6. 'wallet/walletSelected', payload: undefined
7. 'wallet/walletOpened', payload: { id: 4, address: 'oasis1qq2vzcvxn0js5unsch5me2xz4kr43vcasv0d5eq4' }
8. 'wallet/selectWallet', payload: 4

9. 'wallet/walletSelected', payload: undefined
10. 'wallet/walletOpened', payload: { id: 5, address: 'oasis1qq5t7f2gecsjsdxmp5zxtwgck6pzpjmkvc657z6l' }

11. 'wallet/walletSelected', payload: undefined
12. 'wallet/walletOpened', payload: { id: 6, address: 'oasis1qrusst9f8ey6kphgp5d02nuf4kx5qsa80vamtxj8' }

13. 'wallet/walletSelected', payload: undefined
14. 'wallet/walletOpened', payload: { id: 7, address: 'oasis1qpqs54pdwxstpjefw5t6ucy425nt4yud3c4fgr8p' }

15. 'wallet/walletSelected', payload: undefined
16. 'wallet/walletOpened', payload: { id: 8, address: 'oasis1qrf54xw003wkznm90w0q05ns8nryla0455tlex38' }

17. 'wallet/walletSelected', payload: 4
```

But it looks like can fix the bug by just removing `yield* put(walletActions.walletSelected(initialState.selectedWallet))`. There is no reason to select `undefined` when adding every wallet.

File: src/app/state/wallet/saga.test.ts (hunk 1 above)
Comment (buberdds) [reply src/app/state/wallet/saga.test.ts:88]: It's not related to global variable. Test covers another issue when user imports wallets and the first selected wallet exists in redux store. In this case we use existing id not a new one stored in global `walletId` variable. 

Steps to repro (selection modal): 
A - selected
B - selected  
C 
D
E - selected

- nav to import again 
A
B - selected  
C - selected  
D - selected  
E - selected

File: src/app/state/wallet/saga.test.ts (hunk 2 above)
Comment (buberdds) [reply src/app/state/wallet/saga.test.ts:74]: Looks like you have already answered this question :D

**Comment by buberdds** [comment]
Yeah probably there are many ways to fix it. My preference is to select a wallet when prev actions are done.

**Comment by lukaw3d** [comment]
I think we should do:
- commit: test for re-importing existing accounts (this test is already passing before any changes)
- commit: fix the issue by removing `yield* put(walletActions.walletSelected(initialState.selectedWallet))` (selecting `undefined` when adding every wallet makes no sense)
- optionally commit: replacing selectImmediately with a select after all wallets are added

**Review by lukaw3d — APPROVED** [review]

**Comment by buberdds** [comment]
> * (selecting `undefined` when adding every wallet makes no sense)

fyi @lukaw3d this was actually handling a case when you open an active wallet again

**Comment by lukaw3d** [comment]
> > * (selecting `undefined` when adding every wallet makes no sense)
> 
> fyi @lukaw3d this was actually handling a case when you open an active wallet again

Ah, I didn't notice when it was added in https://github.com/oasisprotocol/oasis-wallet-web/pull/787/files#diff-f5039a29f4edb72ce140205418b6a20f84c4e5ed87c38bd33c4724aaee5c66fbR130

when changing from
```ts
  yield* takeLatest(walletActions.selectWallet, selectWallet)
    
  export function* selectWallet({ payload: index }: PayloadAction<number>) {
    ...
    yield* put(push(`/account/${newWallet?.address}`))
  }
```
to
```ts
  useEffect(() => {
    if (typeof activeWalletIndex !== 'undefined' && address) {
      history.push(`/account/${address}`)
    }
  }, [activeWalletIndex, address, history])
```

:/

**Pull request merged**